cocos/platform/CCFileUtils.h

```diff
@@ -874,6 +879,11 @@ class CC_DLL FileUtils
     std::vector<std::string> _searchPathArray;
 
     /**
+     * The search paths which was set by 'setSearchPaths' / 'addSearchPath'.
+     */
+    std::vector<std::string> _originalSearchPaths;
```

Comment:
FileUtils uses `_searchPathArray` for searching full path internally.
`setSearchPaths` will modify `_searchPathArray` variable, it will be connected by `_defaultResourceRootPath`. 
`_originalSearchPaths` saves the original parameter `setSearchPaths` passes in, and `getSearchPaths` returns `_originalSearchPath` will be more reasonable than returning `_searchPathArray`. It's the rule of getter returns what user has set.

If `getSearchPaths` returns `_searchPathArray`, you could suppose the following code will always not work.

If current search path is `bbb/`, `ccc/`
```c++
auto oldDefaultResourceRoot = fileUtils->getDefaultResourceRootPath(); // Suppose it return "" which is an empty string.
auto oldSearchPaths = fileUtils->getSearchPaths(); // oldSearchPaths is : ["bbb/", "ccc/"]
fileUtils->setDefaultResouceRootPath("aaa"); // oldSearchPath is : ["aaa/bbb/", "aaa/ccc"]
// oldSearchPaths will be modified by adding a prefix `aaa` since `_searchPathArray` was modified. 

... ...

// reset resource root
fileUtils->setDefaultResourceRootPath(oldDefaultResourceRoot); 
auto newSearchPath = fileUtils->getSearchPaths(); // newSearchPath is incorrect: ["aaa/bbb", "aaa/ccc"] since it returns `_searchPathArray` which was added by prefix 'aaa'
```